### PR TITLE
added data/text input and refactored forms

### PR DESCRIPTION
**Pull request by stacymcauliffe**
Added design guidelines for text and data input
Made updates to form guidelines to accommodate new information
Removed button alignment information from forms docs

### CODE REVIEW

File: packages/patternfly-4/content/design-guidelines/usage-and-behavior/forms/design.md [hunk 1]
```diff
@@ -10,110 +10,148 @@ Most forms will contain the following basic elements:
 *Form elements*
 ![Form elements](img/form-elements.png)
 
-1. [Labels](/design-guidelines/usage-and-behavior/data-input) - Field labels can be aligned to the left or top of the field depending on the layout of your page and the amount of space you have to work with. Learn more about field labels in the [text and data input design guidelines](/design-guidelines/usage-and-behavior/data-input).
-
-  *Aligned to the top*
+1. [Labels](#labeling) - Field labels can be aligned to the left or top of the field depending on the layout of your page and the amount of space you have to work with.
 
   ![Aligned to the top](img/top-aligned.png)
```
Comment (mcarrano) [inline packages/patternfly-4/content/design-guidelines/usage-and-behavior/forms/design.md]: I wonder whether these two images should be moved to the section below where you talk about labeling. As I read this I expected to find them there.

File: packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# Text input
+Use text input to gather free-form text from a user.
+
+## Elements
+
+![Elements](img/input-elements3.png)
+
+1. Label
+2. Input field
+3. Help
```
Comment (mcarrano) [inline packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md]: Maybe say, "Help can be provided in the form of help text below the field or a popover (infotip). Link to more details below?

File: packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+4. Errors and validation
```
Comment (mcarrano) [inline packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md]: Change to? "Inline Error: Errors resulting from validation of field data should be placed below the field." 

Link to more details below?

File: packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+Helper text is text below a form field that helps a user provide the right information, like Enter a unique name.
+
+**Placeholder text**
+Placeholder text is text inside a form field that provides an example of the required or recommended format for text input. For example, placeholder text for a date might be e.g. DD/MM/YYYY.
+
+For more information about how to provide help content, see  the [forms design guidelines](/design-guidelines/usage-and-behavior/forms#provide-the-right-kind-of-help).
+
+### Errors and validation
+Wherever possible, use inline errors to notify users that something has gone wrong. Inline errors help should be short and help users quickly identify and understand what they need to fix and why.
+![Errors and validation](img/input-error.png)
```
Comment (mcarrano) [inline packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md]: This image is displaying inline with text in the GitHub preview. Try adding a line break before the image.

File: packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+**Helper text**
+Helper text is text below a form field that helps a user provide the right information, like Enter a unique name.
+
+**Placeholder text**
+Placeholder text is text inside a form field that provides an example of the required or recommended format for text input. For example, placeholder text for a date might be e.g. DD/MM/YYYY.
+
+For more information about how to provide help content, see  the [forms design guidelines](/design-guidelines/usage-and-behavior/forms#provide-the-right-kind-of-help).
+
+### Errors and validation
+Wherever possible, use inline errors to notify users that something has gone wrong. Inline errors help should be short and help users quickly identify and understand what they need to fix and why.
```
Comment (mcarrano) [inline packages/patternfly-4/content/design-guidelines/usage-and-behavior/text-input/design.md]: I think that we should mention somewhere that if there is already helper text below the field, the error text will replace it.

**Comment by stacymcauliffe** [comment]
@mcarrano - Thanks for the review! updated content based on your comments.

**Pull request merged**